iceoryx2-ffi/cxx/src/waitset.cpp

```diff
@@ -274,13 +269,13 @@ auto WaitSet<S>::attach_notification(const Listener<S>& listener)
 }
 
 template <ServiceType S>
-auto run_callback(iox2_waitset_attachment_id_h attachment_id, void* context) {
-    auto* fn_call = internal::ctx_cast<iox::function<void(WaitSetAttachmentId<S>)>>(context);
-    fn_call->value()(WaitSetAttachmentId<S>(attachment_id));
+auto run_callback(iox2_waitset_attachment_id_h attachment_id, void* context) -> iox2_callback_progression_e {
```

Comment:
Curious - why is the C enum used? I guess it's because this is called directly by the C API?

Reply:
@orecham this is just the internal implementation. I need a C style callback in the C++ API that has the C++ `iox::function` stored in `context` and can be used as a callback in the C API.